handlers/handlers.go

```diff
@@ -29,60 +37,220 @@ func parseReqBody(w http.ResponseWriter, body io.ReadCloser, bodyObj *models.Con
 	return nil
 }
 
-// PostConversationsHandler creates a new conversation
-func PostConversationsHandler(w http.ResponseWriter, r *http.Request) {
+func getConversation(w http.ResponseWriter, db models.Datastore, id int64) (*models.Conversation, error) {
+	conversation, err := db.GetConversation(id)
+	if err != nil {
+		errMsg := "Internal Server Error"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusInternalServerError)
+		return nil, err
+	}
+
+	if conversation == nil {
+		errMsg := fmt.Sprintf("Conversation %d does not exist", id)
+		log.Println(errMsg)
+		http.Error(w, "Conversation not found", http.StatusNotFound)
+		return nil, nil
+	}
+
+	return conversation, nil
+}
+
+func getMapping(w http.ResponseWriter, db models.Datastore, userID, convID int64) (*models.UserConversationMapping, error) {
+	mapping, err := db.GetUserConversationMapping(userID, convID)
+	if err != nil {
+		errMsg := "Internal Server Error"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusInternalServerError)
+		return nil, err
+	}
+
+	if mapping == nil {
+		errMsg := fmt.Sprintf("User %d is not in conversation %d", userID, convID)
+		log.Println(errMsg)
+		http.Error(w, "Conversation not found", http.StatusNotFound)
+		return nil, err
+	}
+
+	return mapping, nil
+}
+
+// PostConversationHandler creates a single new conversation
+func (env *Env) PostConversationHandler(w http.ResponseWriter, r *http.Request) {
 	defer r.Body.Close()
+	userID, err := strconv.ParseInt(r.Header.Get("User-ID"), 10, 64)
+	if err != nil {
+		errMsg := "Invalid user ID"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusBadRequest)
+		return
+	}
+
+	reqConversation := &models.Conversation{}
+	if err := parseReqBody(w, r.Body, reqConversation); err != nil {
+		return
+	}
 
-	reqBody := &models.Conversation{}
-	if err := parseReqBody(w, r.Body, reqBody); err != nil {
+	if reqConversation.Name == "" || reqConversation.Description == nil {
+		errMsg := "Request body is missing field(s)"
+		fmt.Println(errMsg)
+		http.Error(w, errMsg, http.StatusBadRequest)
 		return
 	}
 
-	json.NewEncoder(w).Encode(reqBody)
+	conversationID, err := env.DB.CreateConversation(reqConversation, userID)
+	if err != nil {
+		errMsg := "Internal Server Error"
+		log.Println(errMsg + ": " + err.Error())
+		http.Error(w, errMsg, http.StatusInternalServerError)
+		return
+	}
+
+	// TODO: Create HTML file
+
+	reqConversation.ID = conversationID
+	location := fmt.Sprintf("/api/conversations/%d", conversationID)
```

Comment:
If we make the `Location` header `/ether/api/conversations/{conversation_id}`, then I think that the routes should also start with `/ether/api/...`, so that there isn't a discrepancy between the two.

Reply:
Hmmm that's true. We can't make the routes have the service name though 'cause then the routing from `heimdall` would get weird, ex `/ether/ether/api/...`. Can you make an issue for `heimdall` then to add the service name to the beginning of the `Location` header of a response if it exists?

Reply:
Does `heimdall` need to remove the service name from the front of any route it receives? Like if it receives a request to `/ether/api/conversations`, can it not just forward to that same path but at `ether`'s location?

Reply:
Oh it doesn't remove the service name from the route though, it just uses the service name as like the DNS nickname thing so if it sends a request to `ether/api/...`, `ether` would receive it as `/api/...`. So if the route was `/ether/api/...` then the client request URL would either have to look like `/ether/ether/api/...` or it would be `/ether/api...` and `heimdall` would have to add `ether` to the start of the request.